src/typechecking/TypeChecker.java

```diff
@@ -721,6 +726,40 @@ private void checkDisjunction(ASTdisjunction disjunction)
 		}
 	}
 
+        /**
+	 * Do typechecking of optimizer statement given by AST node
+	 * 
+	 * @param optimize_statement
+	 *            to be checked
+	 * @throws ParseException
+	 *             if sort violation occurs or undeclared predicate found
+	 */
+	private void checkOptimizeStatement(ASToptimize_statement optimize_statement)
+			throws ParseException {
+		for (int i = 0; i < optimize_statement.jjtGetNumChildren(); i++) {
```

Comment:
I think you can also check here that

(a) optimize parameter list contains exactly one variable
(b) nonrealAtomlist contains exactly one atom, and it is of the form p(X), where X is the variable from the optimize parameter list. I thought we agreed that for now it is better to just have statements of the form:

#minimize{X:p(X)}.

Do you think this will not be sufficient? 

If you add new error checks, it is good to add new test.

I find the more complex things to have unclear semantics, so I would prefer not to have those cases What does it mean if you have #minimize {X:p(X,Y)}, etc?